### PR TITLE
PHPLIB-866: CSFLE prose test for on-demand AWS credentials

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPLIB-866

Patch build: https://spruce.mongodb.com/version/63857b60d6d80a169794c4e0/tasks

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -598,6 +598,18 @@ tasks:
...
             TESTS: "csfle"
 
+    - name: "test-without_aws_credentials"
+      commands:
+        - func: "bootstrap mongo-orchestration"
+          vars:
+            TOPOLOGY: "replica_set"
+        - func: "start kms servers"
+        - func: "run tests"
+          vars:
```
Comment (jmikola) [inline .evergreen/config.yml:608]: I just improved my understanding of the `vars` construct by reading through [Expansions](https://github.com/evergreen-ci/evergreen/wiki/Project-Configuration-Files#expansions) in the Evergreen docs.

> Notice that the function reference can define a set of vars which are treated as expansions within the configuration of the commands in the function.

I reckon my prior confusion was due to our historically inconsistent naming/formatting. I assumed some (like "TESTS") were environment variables, but that's not the case until we explicitly export or declare them when invoking a shell command.

A few more of these PRs and I'll probably want to take on some Evergreen config refactoring.

File: tests/SpecTests/ClientSideEncryptionSpecTest.php [hunk 2]
```diff
@@ -1642,6 +1642,48 @@ static function (self $test, Client $setupClient, ClientEncryption $clientEncryp
...
+
+        $dataKeyOpts = [
+            'masterKey' => [
+                'region' => 'us-east-1',
+                'key' => 'arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0',
+            ],
+        ];
+
+        if (! $shouldSucceed) {
+            $this->expectException(AuthenticationException::class);
```
Comment (jmikola) [inline tests/SpecTests/ClientSideEncryptionSpecTest.php:1676]: This is more specific than some other assertions we make in this file, but I inferred it from the corresponding libmongoc test in https://github.com/mongodb/mongo-c-driver/commit/9c4f49f2e9e69d6ae61070095457fb742a375919#diff-d363a5539f2fbabfadf2d7cc6cdb96a203fb63f2caa31bae731c3e12d97b204cR5013.

File: tests/SpecTests/ClientSideEncryptionSpecTest.php [hunk 3]
```diff
@@ -1642,6 +1642,48 @@ static function (self $test, Client $setupClient, ClientEncryption $clientEncryp
         ];
     }
 
+    /**
+     * Prose test 15: On-demand AWS Credentials
+     *
+     * @see https://github.com/mongodb/specifications/tree/master/source/client-side-encryption/tests#on-demand-aws-credentials
+     * @testWith [true]
+     *           [false]
```
Comment (jmikola) [inline tests/SpecTests/ClientSideEncryptionSpecTest.php:1650]: The `$shouldSucceed=false` case corresponds to testOnDemandAwsCredentials_with_data_set__1 passes in https://spruce.mongodb.com/task/mongo_php_library_test_csfle_without_aws_credentials__php_edge_versions~latest_stable_mongodb_versions~6.0_driver_versions~latest_stable_os~debian11_test_without_aws_credentials_patch_bf91db2558c491f474c986bc50383ff002ec9e01_63857b60d6d80a169794c4e0_22_11_29_03_26_36/tests?execution=0&sortBy=STATUS&sortDir=ASC

File: .evergreen/config.yml (hunk 1 above)
Comment (alcaeus) [reply .evergreen/config.yml:608]: > A few more of these PRs and I'll probably want to take on some Evergreen config refactoring.

I'll make a note of it to remember when the next refactoring inevitably comes around...

**Review by alcaeus — APPROVED** [review]

**Pull request merged**